src/Smalot/PdfParser/Element/ElementHexa.php

```diff
@@ -64,15 +64,21 @@ public static function parse(string $content, ?Document $document = null, int &$
     public static function decode(string $value): string
     {
         $text = '';
-        $length = \strlen($value);
 
-        if ('00' === substr($value, 0, 2)) {
-            for ($i = 0; $i < $length; $i += 4) {
+        // Filter $value of non-hexadecimal characters
+        $value = preg_replace('/[^0-9a-f]/i', '', $value);
```

Comment:
```suggestion
        $value = (string) preg_replace('/[^0-9a-f]/i', '', $value);
```

`preg_replace` returns `null` in case of an error (https://www.php.net/manual/en/function.preg-replace.php). Casting it to `string` would avoid a type error in one of the following string-functions.